lib/src/metta/runner/stdlib.rs

```diff
@@ -1253,6 +1254,64 @@ impl CustomExecute for SubtractionAtomOp {
     }
 }
 
+#[derive(Clone, Debug)]
+pub struct RandomIntOp {}
+
+grounded_op!(RandomIntOp, "random-int");
+
+impl Grounded for RandomIntOp {
+    fn type_(&self) -> Atom {
+        Atom::expr([ARROW_SYMBOL, ATOM_TYPE_NUMBER, ATOM_TYPE_NUMBER, ATOM_TYPE_NUMBER])
+    }
+
+    fn as_execute(&self) -> Option<&dyn CustomExecute> {
+        Some(self)
+    }
+}
+
+impl CustomExecute for RandomIntOp {
+    fn execute(&self, args: &[Atom]) -> Result<Vec<Atom>, ExecError> {
+        let arg_error = || ExecError::from("random-int expects two arguments: number (start) and number (end)");
+        let start = AsPrimitive::from_atom(args.get(0).ok_or_else(arg_error)?).as_number().ok_or_else(arg_error)?;
+        let end = AsPrimitive::from_atom(args.get(1).ok_or_else(arg_error)?).as_number().ok_or_else(arg_error)?;
+        let range = Into::<i64>::into(start)..Into::<i64>::into(end);
+        if range.is_empty() {
+            return Err(ExecError::from("Wrong range"));
+        }
+        let mut rng = rand::thread_rng();
+        Ok(vec![Atom::value(rng.gen_range::<i64,_>(range))])
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct RandomFloatOp {}
+
+grounded_op!(RandomFloatOp, "random-float");
+
+impl Grounded for RandomFloatOp {
+    fn type_(&self) -> Atom {
+        Atom::expr([ARROW_SYMBOL, ATOM_TYPE_NUMBER, ATOM_TYPE_NUMBER, ATOM_TYPE_NUMBER])
+    }
+
+    fn as_execute(&self) -> Option<&dyn CustomExecute> {
+        Some(self)
+    }
+}
+
+impl CustomExecute for RandomFloatOp {
+    fn execute(&self, args: &[Atom]) -> Result<Vec<Atom>, ExecError> {
+        let arg_error = || ExecError::from("random-float expects two arguments: number (start) and number (end)");
+        let start = AsPrimitive::from_atom(args.get(0).ok_or_else(arg_error)?).as_number().ok_or_else(arg_error)?;
+        let end = AsPrimitive::from_atom(args.get(1).ok_or_else(arg_error)?).as_number().ok_or_else(arg_error)?;
+        let range = Into::<f64>::into(start)..Into::<f64>::into(end);
+        if range.is_empty() {
+            return Err(ExecError::from("Wrong range"));
+        }
+        let mut rng = rand::thread_rng();
+        Ok(vec![Atom::value(rng.gen_range::<f64,_>(range))])
```

Comment:
Version `0.9.0` of `rand` changes api by adding `Result` into `SampleRange::sample_single` signature (see [docs](https://docs.rs/rand/0.9.0-alpha.2/rand/distr/uniform/trait.SampleRange.html)). Thus it will be possible to do `match range.sample_single(rng) {` and check the result of sampling. It will cover overflow error as well as some others. I would suggest adding comment about it and waiting for the library upgrade.